### ISSUE TITLE
Improve API of alert component

### ISSUE THREAD

**Issue by chausknecht** [jwstegemann/fritz2#292]
**Note:** See [PR#361](https://github.com/jwstegemann/fritz2/pull/361) for breaking changes!

``alert`` offers an extension function on ``ComponentValidationMessage`` in order ease the process of emebdding messages into the UI. Currently this approach is used within our components by the ``formControl`` component, which has kinda dominated the API for its own needs, so it lacks important possibilities:

- the parameters should reflect *most* if not *all* *properties* of the ``AlertComponent``, besides the content and severity, which is defined by the message itself. All other aspects should be customizable!
- to avoid code duplication there should be some smart approach how to *reuse* the AlertComponent in a way.
- the current ``severity`` property of alert should be made private and hidden from the outside world.

**Comment by haukesomm** [comment]
So I just stumbled over this issue and thought I might as well take care of it 😄 

> the parameters should reflect most if not all properties of the AlertComponent, besides the content and severity, which is defined by the message itself. All other aspects should be customizable!

> the current severity property of alert should be made private and hidden from the outside world.

Good point! I already thought something similar and I think we can get rid of the AlertComponent's `severity` property altogether by offering two new and more flexible properties instead: `color` (which should utilize the new `ColorScheme` class) and `icon`. 
This enables us to remove the irritating and unflexible `AlertSeverity` class from the theme and be more flexible with custom styling, both via the standard factory method as well as the `ComponentValidationMessage`-extension.

> to avoid code duplication there should be some smart approach how to reuse the AlertComponent in a way.

Is there anything in particular the could be done in your opinion?

**Comment by haukesomm** [comment]
> the parameters should reflect most if not all properties of the AlertComponent, besides the content and severity, which is defined by the message itself. All other aspects should be customizable!

The `AlertComponent` needs some more changes in order to allow the above mentionend customization.

**Comment by chausknecht** [comment]
The current improvement only deals with the central theming aspects.

The main focus of this issue is the offering of *configuration parameters* (like the styling) for the ``ComponentValidationMessage.asAlert``! This still lacks.

Have a look at the [TicTacToe]() example and the usage of the ``alert`` component for game messages like the winning or other ending messages. As there is a need to style the alert a little, it currently must be built by hand:
```kotlin
alert({
    // needed
    width { full }
    radius { small }
}) {
    // would be done by ``.asAlert()`` method, so it is cumbersome to craft this by hand
    severity { if (it.messages.first().severity == Severity.Success) success else info }
    content(it.messages.first().message)
}
```

It would be good if this could be less cumbersome:
```kotlin
messages.first().asAlert({
    width { full }
    radius { small }
})
```

So imho the issue is still not fixed with version 0.10. and PR #356.

**Comment by haukesomm** [comment]
Agreed, it was my fault to overlook this part of the issue. I opened PR #361 to work on this.
